main.py

```diff
@@ -157,15 +136,22 @@ def main(
             flatfields.append(basic.flatfield)
             darkfields.append(basic.darkfield)
 
-    output_folder = Path(output_folder)
-    input_path2 = get_main_name(input_path)
-    flatfield_path = output_folder / (input_path2 + "-ffp.tiff")
-    darkfield_path = output_folder / (input_path2 + "-dfp.tiff")
+    # Re-arrange flatfields and darkfields axis to be (z, y, x, c)
     flatfields = np.moveaxis(np.array(flatfields), 0, -1)
     darkfields = np.moveaxis(np.array(darkfields), 0, -1)
+
+    # Get output file names, splitext gets the file name without the extension
+    flatfield_path = args.output_folder / f"{splitext(args.input.name)[0]}-ffp.tiff"
```

Comment:
This gives a slightly different output from the original version, but I believe it's OK.